caikit/runtime/utils/import_util.py

```diff
@@ -161,6 +162,8 @@ def get_dynamic_module(module_name: str, module_dir: str = None) -> ModuleType:
     Returns:
         (module): Handle to the module after dynamic import
     """
+    if module := sys.modules.get(module_name):
```

Comment:
This is needed to short-circuit for the "special" module `__main__` that is already (always) in `sys.modules`, but will fail on `importlib.util.find_spec` since there is not spec to be found for it.

Reply:
It also provides a nice quick-escape for modules that have already been imported some other way

Reply:
Nice!